### DISCUSSION TITLE
Home key lock without MQTT

### DISCUSSION THREAD

**Discussion by Maxime0506** [rednblkx/HomeKey-ESP32#53] [Show and tell]
Hello,
I am currently working on a project where I want to control a relay that is directly connected to the esp32 and doesn't require MQTT or Home assistant

When I am finished I am going to upload the edited code here in the discussion fell free to use it in your own project.

**Comment by Maxime0506** [comment]
I have uploaded my edited version of the code. I removed mqtt and added a relay that stays on for 5 seconds and then closes:
https://github.com/Maxime0506/HomeKey-ESP32-NO-MQTT

**Reply by smolleman** [reply]
The link doesn't work, this is the correct one: [https://github.com/Maxime0506/HomeKey-ESP32-NO-MQTT](https://github.com/Maxime0506/HomeKey-ESP32-NO-MQTT)

**Reply by dqduc** [reply]
I saw quite some updates from main repo. Is it possible to keep your repo up to date ? It's useful for no-mqtt users

**Reply by Maxime0506** [reply]
I have an idea in mind, but the challenge is that I haven’t had the time to fully test it yet. I need to remove MQTT entirely and rebuild from scratch, as the updated code has a completely new structure. This will take some time, but I’ll make it a priority because I also need the updated version. I’ll aim to have it done soon.
